slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/InterchangeV5Importer.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection.interchange;
+
+import tech.pegasys.web3signer.slashingprotection.dao.SignedAttestationsDao;
+import tech.pegasys.web3signer.slashingprotection.dao.SignedBlocksDao;
+import tech.pegasys.web3signer.slashingprotection.dao.Validator;
+import tech.pegasys.web3signer.slashingprotection.dao.ValidatorsDao;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.Metadata;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.SignedAttestation;
+import tech.pegasys.web3signer.slashingprotection.interchange.model.SignedBlock;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.tuweni.bytes.Bytes;
+import org.jdbi.v3.core.Jdbi;
+
+public class InterchangeV5Importer {
+
+  private static final int FORMAT_VERSION = 5;
+
+  private final Jdbi jdbi;
+  private final ValidatorsDao validatorsDao;
+  private final SignedBlocksDao signedBlocksDao;
+  private final SignedAttestationsDao signedAttestationsDao;
+  private final ObjectMapper mapper;
+
+  public InterchangeV5Importer(
+      final Jdbi jdbi,
+      final ValidatorsDao validatorsDao,
+      final SignedBlocksDao signedBlocksDao,
+      final SignedAttestationsDao signedAttestationsDao,
+      final ObjectMapper mapper) {
+    this.jdbi = jdbi;
+    this.validatorsDao = validatorsDao;
+    this.signedBlocksDao = signedBlocksDao;
+    this.signedAttestationsDao = signedAttestationsDao;
+    this.mapper = mapper;
+  }
+
+  public void importData(final InputStream input) throws IOException {
+
+    try (JsonParser jsonParser = mapper.getFactory().createParser(input)) {
+      final ObjectNode rootNode = mapper.readTree(jsonParser);
+
+      final JsonNode metadataJsonNode = rootNode.get("metadata");
+      final Metadata metadata = mapper.treeToValue(metadataJsonNode, Metadata.class);
+
+      if (metadata.getFormatVersion() != FORMAT_VERSION) {
+        throw new IllegalStateException("Expecting an interchange_format_version of 5");
```

Comment:
nit: `"Expecting an interchange_format_version of + FORMAT_VERSION"`

Reply:
done